test/jdk/javax/swing/JComboBox/DisabledComboBoxFontTest.java

```diff
@@ -0,0 +1,138 @@
+            "editable JComboBoxes and two JLists displayed correctly when " +
+            "enabled and disabled. Fail if they don't. Toggle UI enabled " +
+            "status with the button on the right of the frame.";
+
+    private static JFrame frame;
+    private static boolean hasNimbus;
+
+    public static void createAndShowGUI() throws InterruptedException,
+            InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+
+            JComboBox combo = new JComboBox();
+            combo.addItem("Simple JComboBox");
+            combo.addItem("Simple JComboBox2");
+            combo.setEnabled(false);
+
+            JComboBox customCombo = new JComboBox();
+            customCombo.setRenderer(new DefaultListCellRenderer());
+            customCombo.addItem("JComboBox with DefaultListCellRenderer");
+            customCombo.addItem("JComboBox with DefaultListCellRenderer2");
+            customCombo.setEnabled(false);
+
+            String[] s = {"one", "two", "three"};
+            JList list = new JList(s);
+            list.setEnabled(false);
+            JList list2 = new JList(s);
+            list2.setCellRenderer(new DefaultListCellRenderer());
+            list2.setEnabled(false);
+
+            JButton btn = new JButton("Enable/Disable");
+            ActionListener actionListener = event -> {
+                combo.setEnabled(!combo.isEnabled());
+                customCombo.setEnabled(!customCombo.isEnabled());
+                list.setEnabled(!list.isEnabled());
+                list2.setEnabled(!list2.isEnabled());
+                String str = event.getActionCommand();
+                System.out.println("Clicked = " + str + " " + customCombo.isEnabled());
+            };
+            btn.addActionListener(actionListener);
+
+            FlowLayout layout = new FlowLayout(FlowLayout.LEADING);
+            JPanel panel = new JPanel(layout);
+            panel.add(combo);
+            panel.add(customCombo);
+            panel.add(list);
+            panel.add(list2);
+            panel.add(btn);
+            System.out.println("RENDERER1: " + combo.getRenderer());
+            System.out.println("RENDERER2: " + customCombo.getRenderer());
+            System.out.println("RENDERER3: " + list.getCellRenderer());
+            System.out.println("RENDERER4: " + list2.getCellRenderer());
+
+            frame = new JFrame();
+            frame.getContentPane().add(panel);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+
+            frame.setVisible(true);
+        });
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        hasNimbus = false;
+
+        for (LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
```

Comment:
It seems the intention to run the test is in Nimbus LAF only. I think you can directly set the Nimbus LAF using `UIManager.setLookAndFeel("javax.swing.plaf.nimbus.NimbusLookAndFeel");` and can avoid the `for-loop, condition check` and `hasNimbus `variable.

To print the error message, LAF can be set within try-catch block.

Reply:
Thanks for the suggestion. I made the change. I initially had this loop to test locally for all LAF's, but ended up using this test for Nimbus only as originally designed.